UnityProject/Assets/Scripts/Input System/CablePlacementVisualisation.cs

```diff
@@ -0,0 +1,239 @@
+			SetConnectionPointColor(startPoint.x, startPoint.y, startPointColor);
+		}
+		// if mouse up - stop drawing and check if can build
+		else if (CommonInput.GetMouseButtonUp(0))
+		{
+			endPoint = currentlyHoveredPoint;
+			// TODO: check if correct and build
+			Build();
+			ResetValues();
+		}
+
+		// check if startPoint has changed (-Vector2Int.one is default value)
+		if (startPoint != -Vector2Int.one)
+		{
+			lineRenderer.SetPositions(new Vector3[]
+			{
+				connectionPoints[startPoint.x, startPoint.y].transform.localPosition,                        // position of start point
+                connectionPoints[currentlyHoveredPoint.x, currentlyHoveredPoint.y].transform.localPosition   // position of current point
+            });
+		}
+	}
+
+	/// <summary>
+	/// Try to build cable
+	/// </summary>
+	private void Build()
+	{
+		if (startPoint == endPoint || Mathf.Abs(startPoint.x - endPoint.x) > 1 || Mathf.Abs(startPoint.y - endPoint.y) > 1) return;
+
+		GameObject target = MouseUtils.GetOrderedObjectsUnderMouse().FirstOrDefault();
+		Vector2 targetVector = new Vector2(startPoint.x - endPoint.x, startPoint.y - endPoint.y);
+
+
+		Connection start = GetConnectionByPoint(startPoint);
+		Connection end = GetConnectionByPoint(endPoint);
+		CableApply cableApply = CableApply.ByLocalPlayer(target, start, end, Vector2Int.RoundToInt(transform.position + new Vector3(0.5f, 0.5f, 0)));
+
+		//if HandObject is null, then its an empty hand apply so we only need to check the receiving object
+		if (cableApply.HandObject != null)
+		{
+			//get all components that can contains CableApply interaction
+			var cableAppliables = cableApply.HandObject.GetComponents<MonoBehaviour>()
+				.Where(c => c != null && c.enabled && (c is IBaseInteractable<CableApply>));
+
+			foreach (var cableAppliable in cableAppliables.Reverse())
+			{
+				var hap = cableAppliable as IBaseInteractable<CableApply>;
+				if (hap.ClientCheckAndTrigger(cableApply)) return;
+			}
+		}
+	}
+
+	/// <summary>
+	/// Reset startPoint, endPoint and lineRenderer
+	/// </summary>
+	private void ResetValues()
+	{
+		// reset line renderer
+		lineRenderer.SetPositions(new Vector3[]
+		{
+			Vector3.zero,
+			Vector3.zero
+		});
+
+		// reset colors
+		foreach (var point in connectionPoints)
+			point.color = defaultPointColor;
+
+		// reset points
+		startPoint = -Vector2Int.one;
+		endPoint = -Vector2Int.one;
+	}
+
+	/// <summary>
+	/// Change color of connection point
+	/// </summary>
+	/// <param name="x">x position</param>
+	/// <param name="y">y position</param>
+	/// <param name="color">target color</param>
+	private void SetConnectionPointColor(int x, int y, Color color)
+	{
+		// get sprite renderer
+		SpriteRenderer spriteRenderer = connectionPoints[x, y];
+		// change color
+		spriteRenderer.color = color;
+	}
+	/// <summary>
+	/// Convert Vector2Int point to Connection based on position
+	/// (point values should be in range from 0 to <see cref="maxIndex"/>)
```

Comment:
This comment doesn't make sense as far as I can see. maxIndex is basically coupled to the Connection enum which means it can't be changed - a maxIndex of 10 has no meaning as the rest of the code wont work unless Connection is defined I think?

Reply:
fixed (9ef7753f67ad74949d5bfa8329a959a114742c88)